### ISSUE TITLE
test_arch_compat asserts an unseeded random draw against a one-sided 0.011 margin at the upper bound

### ISSUE THREAD

**Issue by cklxx** [acupof-ai/aupai#582]
Found 2026-09-19 while attributing two CI reds on PR #580 (a datagen-only PR that touches neither file). Not caused by that PR; the assert and the init it reads are both months old.

## The shape

`scripts/test_arch_compat.py:186-188` (unchanged since `aa3a8b96`, 2026-08-26):

```python
dt_bias = m.blocks[0].mixer.dt_bias
ret = torch.exp(-torch.nn.functional.softplus(dt_bias)).mean().item()
assert 0.85 < ret < 0.99, ret
```

`dt_bias` comes from `model.py:138` (unchanged since `be845ec0`):

```python
dt = torch.exp(torch.rand(self.h * self.hd) * (math.log(0.1) - math.log(1e-3)) + math.log(1e-3))
self.dt_bias = nn.Parameter(dt + torch.log(-torch.expm1(-dt)))
```

`torch.rand` — **unseeded**. No `manual_seed` precedes line 185's block (`m` is built at :164 for the optimizer-plumbing block; the nearest preceding seed is at :21x, after). The comment above the init says "mean retention ~0.9 per token", but that is the **median**: the true mean of `exp(-dt)` for `dt ~ logU[1e-3, 0.1]` is **0.9790**.

So the asserted window is one-sided and barely clears the init: **0.011 of headroom above** (0.99 − 0.9790) versus **0.129 below** (0.9790 − 0.85).

## Measured

At the check's own config (`Cfg.d=64, heads=2` at :89 → `h*hd = 64` draws per build):

- 200 consecutive unseeded builds: min 0.97068, max 0.98729, mean 0.97879, **0/200 outside the window**
- 500k simulated draws: mean 0.97904, sd 0.00300; **P(mean > 0.99) ≈ 1e-5** (a ~1-in-125,000 build); observed max 0.99077

Observed in CI: **0.9904198**, i.e. **3.8 sigma**.

## The failure is real, the instrument is wrong

A CI run failed on this assert (`run 35427256949`). Rerunning the same sha with no code change, the same job **passed** (`attempt=2`, success) — which is the signature of a draw, not of a regression.

Two things are worth separating:
1. The check is *flaky by construction*: it asserts a statistic of an unseeded draw, so a green is not evidence the init is right and a red is not evidence it is wrong.
2. The init and its documentation disagree. "~0.9" is a median quoted as a mean. Anyone reasoning about retention from that comment is off by 0.08.

## Candidate fixes (any one; not all)

- **Seed it** before the block, so the assert becomes deterministic and the flake disappears. Cheapest, but the assert then only ever tests one draw.
- **Assert against the init's real distribution** — e.g. assert the per-parameter `dt` lies in `logU[1e-3, 0.1]` (the property the init is designed to have), rather than a mean with a knife-edge bound.
- **Widen the upper bound** past the init's real support (~0.999 per-draw max) if the mean is what must be pinned — but then the assert stops discriminating much.
- Correct the comment to say median, and state the mean.

Recommend the second: it asserts the property the code intends and does not depend on the RNG.

Not in scope for #553 / PR #580; opened separately per controller ruling.
